src/sentry/static/sentry/app/components/sidebar/broadcasts.jsx

```diff
@@ -124,20 +124,13 @@ const Broadcasts = createReactClass({
     let unseenBroadcastIds = this.getUnseenIds();
     if (unseenBroadcastIds.length === 0) return;
 
-    this.api.request('/broadcasts/', {
-      method: 'PUT',
-      query: {id: unseenBroadcastIds},
-      data: {
-        hasSeen: '1',
-      },
-      success: () => {
-        this.setState({
-          broadcasts: this.state.broadcasts.map(item => {
-            item.hasSeen = true;
-            return item;
-          }),
-        });
-      },
+    broadcastActions.markAsSeen(this.api, unseenBroadcastIds).then(data => {
+      this.setState({
```

Comment:
We should use the callback version of `setState` when referencing current state:

`this.setState(state => state.broadcasts.map(item => {...}`